src/service/pluginService.ts

```diff
@@ -127,210 +112,147 @@ export class PluginService {
+    const missingExtensions = PluginService.GetMissingExtensions(
       extensions,
       ignoredExtensions
     );
-    if (missingList.length === 0) {
+    if (missingExtensions.length === 0) {
       notificationCallBack("Sync : No Extensions needs to be installed.");
       return [];
     }
-    addedExtensions = await PluginService.ProcessInstallationCLI(
-      missingList,
-      osType,
-      insiders,
+    addedExtensions = await PluginService.InstallWithAPI(
+      missingExtensions,
       notificationCallBack
     );
     return addedExtensions;
   }
 
-  public static async ProcessInstallationCLI(
-    missingList: ExtensionInformation[],
-    osType: OsType,
-    isInsiders: boolean,
+  public static async InstallWithAPI(
+    missingExtensions: ExtensionInformation[],
     notificationCallBack: (...data: any[]) => void
   ): Promise<ExtensionInformation[]> {
-    const addedExtensions: ExtensionInformation[] = [];
-    const exec = require("child_process").exec;
-    notificationCallBack("TOTAL EXTENSIONS : " + missingList.length);
-    notificationCallBack("");
-    notificationCallBack("");
-    let myExt: string = process.argv0;
-    console.log(myExt);
-    let codeLastFolder = "";
-    let codeCliPath = "";
-    if (osType === OsType.Windows) {
-      if (isInsiders) {
-        codeLastFolder = "Code - Insiders";
-        codeCliPath = "bin/code-insiders";
-      } else {
-        codeLastFolder = "Code";
-        codeCliPath = "bin/code";
-      }
-    } else if (osType === OsType.Linux) {
-      if (isInsiders) {
-        codeLastFolder = "code-insiders";
-        codeCliPath = "bin/code-insiders";
-      } else {
-        codeLastFolder = "code";
-        codeCliPath = "bin/code";
-      }
-    } else if (osType === OsType.Mac) {
-      codeLastFolder = "Frameworks";
-      codeCliPath = "Resources/app/bin/code";
-    }
-    myExt =
-      '"' +
-      myExt.substr(0, myExt.lastIndexOf(codeLastFolder)) +
-      codeCliPath +
-      '"';
-    for (let i = 0; i < missingList.length; i++) {
-      const missExt = missingList[i];
-      const name = missExt.publisher + "." + missExt.name;
-      const extensionCli = myExt + " --install-extension " + name;
-      notificationCallBack(extensionCli);
-      try {
-        const installed = await new Promise<boolean>(res => {
-          exec(extensionCli, (err, stdout, stderr) => {
-            if (!stdout && (err || stderr)) {
-              notificationCallBack(err || stderr);
-              res(false);
-            }
-            notificationCallBack(stdout);
-            res(true);
+    return new Promise((resolve, reject) => {
+      const addedExtensions: ExtensionInformation[] = [];
+      let remainingExtensions: ExtensionInformation[] = [...missingExtensions];
+
+      notificationCallBack("TOTAL EXTENSIONS : " + missingExtensions.length);
+      notificationCallBack("");
+      notificationCallBack("");
+      missingExtensions.forEach(async ext => {

```

Comment:
and use `await` here.

Reply:
Again, I don't really know how to use await here so I just used `Promise.all` to wait for them to finish.